status/pricing.py

```diff
@@ -605,24 +605,79 @@ def get(self):
                               user=self.get_current_user()))
 
 
-class GenerateQuoteHandler(SafeHandler):
-    """ Serves a built preliminary pricing quote page
+class AgreementsDBHandler(SafeHandler):
+    """Base class that other handlers can inherit from to access the Agreements DB.
+
+    Implements logic that other classes can reuse
+    """
+
+    # _____________________________ FETCH METHODS _____________________________
+
+    def fetch_agreement(self, project_id):
+        db = self.application.agreements_db
+        rows = db.view('entire_document/project_id', descending=True, key=project_id, limit=1).rows
+        doc = {}
+        if len(rows)>0:
+            doc = rows[0].value
+        return doc
+
+    # ____________________________ UPDATE DOC METHODS ____________________
+
+    def update_agreementdoc(self, project_id, timestamp, save_info=None):
+        doc = self.fetch_agreement(project_id)
+        #for saving agreement info
+        if save_info:
+            if not doc:
+                doc['project_id'] = project_id
+            if 'saved_agreements' in doc:
+                doc['saved_agreements'][timestamp] = save_info
+            else:
+                doc['saved_agreements'] = { timestamp: save_info }
+        #for marking a saved agreement signed
+        else:
+            doc['signed'] = timestamp
+            doc['signed_by'] = self.get_current_user().name
+            doc['signed_at'] = int(datetime.datetime.now().timestamp()*1000)
```

Comment:
We should probably think about timezones and DST and so on here, haven't really implemented it yet. I hope it would go easier if we save the epoch time.

Reply:
Yeah, I would say save as UTC (I think epoch time is always UTC?) and display as local time.